src/pages/SupplyRequisitionPage.js

```diff
@@ -0,0 +1,362 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2016
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import autobind from 'react-autobind';
+import { View } from 'react-native';
+
+import { GenericPage } from './GenericPage';
+import globalStyles from '../globalStyles';
+import { formatDate, parsePositiveInteger, sortDataBy } from '../utilities';
+import { createRecord } from '../database';
+import { buttonStrings, modalStrings, pageInfoStrings, tableStrings } from '../localization';
+import {
+  AutocompleteSelector,
+  PageButton,
+  PageInfo,
+  PageContentModal,
+  TextEditor,
+} from '../widgets';
+
+const DATA_TYPES_SYNCHRONISED = ['RequisitionItem', 'Item', 'ItemBatch'];
+const MODAL_KEYS = {
+  COMMENT_EDIT: 'commentEdit',
+  ITEM_SELECT: 'itemSelect',
+  MONTHS_SELECT: 'monthsSelect',
+};
+
+export class SupplyRequisitionPage extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      modalKey: null,
+      modalIsOpen: false,
+      selection: [],
+    };
+    this.dataFilters = {
+      searchTerm: '',
+      sortBy: 'itemName',
+      isAscending: true,
+    };
+    autobind(this);
+  }
+
+  /**
+   * Respond to the user editing the number in the required quantity column
```

Comment:
Might be worth noting here that it is updating the related customer invoice too.